extensions/amp-analytics/0.1/activity-impl.js

```diff
@@ -124,7 +124,9 @@ const ACTIVE_EVENT_TYPES = [
  * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampDoc
  */
 export function installActivityServiceForTesting(ampDoc) {
-  registerServiceBuilderForDoc(ampDoc, 'activity', Activity);
+  registerServiceBuilderForDoc(ampDoc,
+      'activity',
+      ampdoc => new Activity(ampdoc));
```

Comment:
The actual P0 ended up due to timing. Overall, the foreseen use of constructor-as-factory is fine, unforeseen could easily cause problems. Obviously, the most dangerous case would be reverse - when a constructor is used as a factory. So making them one and the same is better for long-term.